app/helpers/katello/katello_urls_helper.rb

```diff
@@ -1,15 +1,23 @@
 require 'uri'
 module Katello
   module KatelloUrlsHelper
+    FOREMAN_SETTING_URL = Setting[:foreman_url].sub(/^https/, 'http').freeze
+
     def subscription_manager_configuration_url(host = nil, rpm = true)
       prefix = if host && host.content_source
-                 "http://#{@host.content_source.hostname}"
+                 "http://#{host.content_source.hostname}"
                else
-                 Setting[:foreman_url].sub(/\Ahttps/, 'http')
+                 FOREMAN_SETTING_URL
                end
       config = rpm ? SETTINGS[:katello][:consumer_cert_rpm] : SETTINGS[:katello][:consumer_cert_sh]
 
       "#{prefix}/pub/#{config}"
     end
+
+    def repository_url(content_path)
+      return content_path if content_path.start_with?('http', 'https', 'ftp', 'ftps')
+      return "#{FOREMAN_SETTING_URL}/#{content_path}" unless @host && @host.content_source
```

Comment:
There is still missing the part around the lifecycle environment and `self.repo_path_from_content_path`, that we talked about last time.

Reply:
opps, thanks, adding it